WebDriverAgentLib/Commands/FBSessionCommands.m

```diff
@@ -123,6 +123,11 @@ + (NSArray *)routes
     FBConfiguration.waitForIdleTimeout = [capabilities[FB_SETTING_WAIT_FOR_IDLE_TIMEOUT] doubleValue];
   }
 
+  if (nil == capabilities[FB_SETTING_FORCE_TURN_ON_SOFTWARE_KEYBOARD] ||
```

Comment:
cannot we reuse the existing `connectHardwareKeyboard` setting?

Reply:
I initially thought the point but I thought simply 'respect current simulator preference' was better.
For example, via GUI, turn on/off software keyboard works while hardware keyboard can keep on.

So, I thought "keeping current pre-set simulator preference" should allow users rather than hardware keyboard vs software keyboard (via this PR's keys). It can do with `connectHardwareKeyboard: false` and `forceTurnOnSoftwareKeyboard: false`